src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -927,7 +939,7 @@ public void Dispose()
         // Handle Exited event and display process information.
         private void myProcess_Exited(object sender, System.EventArgs e)
         {
-            if (System.Threading.Interlocked.Decrement(ref _numberOfProcessesToWaitFor) == 0)
+            if ((System.Threading.Interlocked.Decrement(ref _numberOfProcessesToWaitFor) == 0) || Any)
```

Comment:
I guess we can remove prefix `System.Threading` and maybe swap conditions.
```suggestion
            if (Any || (Interlocked.Decrement(ref _numberOfProcessesToWaitFor) == 0))
```